### PR TITLE
Some icon refactors and old sprite hemmoraging, part 1 of 2

### PR DESCRIPTION
**Pull request by ghost**
- Splits guns.dmi up into projectile.dmi, magic.dmi, and energy.dmi
- Cleans out a ton of unused under clothing icons
- Removes purple lawyer jacket and suit, replaces with tan jacket and suit
- Removes a few 'suit jacket' under suits (very poorly sprited, and formal should be a combination of suit slot + under slot) (burgundy, checkered, etc) (the female executive suit and black suit with the red tie still remain though, they're pretty slick looking)
- Removes ties from lawyer and detective suits
- Makes them spawn with actual ties and puts ties in their lockers
- Removes 'blue' pants (they looked identical to jeans)
- Removes johnny five aces jacket and undersuit
- Renames ba_suit to bar_suit (stop naming icon states so poorly thanks)
- Removes justice suit
- Moves winter coats to the back of heads of staff lockers and removes santa hats (winter coats are a LOT less prolific now that it is no longer the holidays)

I'm not going to refactor clothing in this round, as I think the main issue was a lot of bloat of unused assets. Gun refactoring was 100% necessary though considering energy guns use like 10 icons each to show their charge amount (this should be converted to overlays.)

### CODE REVIEW

**Comment by paprka** [comment]
Inhand refactor tomorrow.

File: code/modules/clothing/suits/jobs.dm [hunk 1]
```diff
@@ -107,11 +107,11 @@
...
 	togglename = "buttons"
 
-/obj/item/clothing/suit/toggle/lawyer/purple
+/obj/item/clothing/suit/toggle/lawyer/tan
 	name = "purple suit jacket"
 	desc = "A foppish dress jacket."
-	icon_state = "suitjacket_purp"
-	item_state = "suitjacket_purp"
+	icon_state = "suitjacket_tan"
+	item_state = "suitjacket_tan"
```
Comment (Incoming5643) [inline code/modules/clothing/suits/jobs.dm]: It's still called a purple suit jacket even though it's apparently tan?

Comment (paprka) [inline code/modules/clothing/suits/jobs.dm]: missed those two lines, yeah

**Comment by paprka** [comment]
Holy shit github, stop being fucking broken.

**Comment by TZK13** [comment]
Is the reasoning for removing the purple lawyer jacket and suit because of it being a Miles Edgeworth reference? I mean the blue lawyer outfit is also a reference to Phoenix Wright's. I can't say this really bothers me very much considering I never liked how puffy the purple one looked but they were hardly bad sprites.

**Comment by paprka** [comment]
A lawyer wearing a blue jacket and blue slacks is so much more believable than purple, but it's not necessarily a hard thing to change if we want our lawyers in typical black/tan

Blue suits are actually fairly common

**Comment by TZK13** [comment]
On the one hand I do agree that the purple outfit did look a little stupid and goofy but that's just my personal opinion, there's certainly going to be people that disagree with that and who will use this as more ammo against you and against coderbus as well because you know people will jump to the copyright argument. I also don't think it's a good idea to remove something like that in a PR titled merely as refactoring icons, people have certainly accused you of hiding little removals and changes in larger PRs before. I won't accuse you of intentionally trying to do that here but it certainly doesn't seem worth it in my opinion.

**Comment by TZK13** [comment]
Also another thing, in removing the tie that comes with the grey detective suit you can't do the waistcoat look pictured in #6779 because you can't wear multiple accessories on your uniform at once. Just saying, it's a rather nice and iconic detective look to have a waistcoat over a suit with tie, being able to wear multiple accessories at once would be a nice thing to look at maybe.

**Comment by paprka** [comment]
waistcoats are for waiters, and the bartender's uniform comes with a bowtie for this reason

if this really bothers you, add a way to add multiple accessories to a uniform, but it's not my problem because i don't see that as a feature critical to the way the detective functions compared to every dress shirt using the new tie system instead of painted on ties

nobody will complain about the miles edgeworth costime being removed, and if they do, nobody here gives a shit

**Comment by paprka** [comment]
i can't fix the map here, at all, and i can't figure out why

i literally downloaded the map from /tg/'s master and it's still saying i have over 6000 line changes

gg

**Comment by paprka** [comment]
I'll clean these files when the map merger doesn't suck or whatever

**Comment by tkdrg** [comment]
Can't you have different PRs, one fixing the inhand issue, the other removing things you don't like?

**Comment by paprka** [comment]
that's what this is though, this has nothing to do with inhands

**Comment by AnonymousNow** [comment]
If I'm reading this correctly, you've pre-emptively dismissed the opinions of people disapproving of the purple suit removal - and informed people that there's nothing they can do, they can just Deal With It.

**Comment by paprka** [comment]
Anonymousnow you can contribute to the code at any time fuckboy

**Comment by xxalpha** [comment]
This PR is shit, don't remove items people like and have no effect on gameplay.

**Comment by paprka** [comment]
welp time to close the comments

**Pull request closed without merging**